vmdb/app/controllers/application_controller/miq_request_methods.rb

```diff
@@ -292,12 +292,12 @@ def build_pxe_img_grid(pxe_imgs, sort_order = nil, sort_by = nil)
   end
 
   def build_iso_img_grid(iso_imgs,sort_order="ASC",sort_by="name")
-    @edit[:iso_img_sortdir] = sort_order
     @edit[:iso_img_headers] = {
         "name"=>"Name"
-    }                                                         #Using it to get column display name on screen to show sort by
-    @edit[:iso_img_columns] = ["name"] #Using it to get column names for sort
-    @edit[:iso_img_sortcol] = sort_by                 # in case sort column is not set, set the defaults
+    }                                                       # Using it to get column display name on screen to show sort by
+    @edit[:iso_img_columns] = ["name"]                      # Using it to get column names for sort
+    @edit[:iso_img_sortcol] = sort_by    ||= "name"         # in case sort column is not set, set the defaults
+    @edit[:iso_img_sortdir] = sort_order ||= "ASC"
```

Comment:
I'd say that whole file is a bit of a problem :). But yes, I'd like to eventually see it cleaned up a bit more but I've at least tried to unify these a bit so that the common functionality is in just one place.

Please do review the latest commit, I'm *almost* sure it should be semantically equivalent (except possibly for the order of entries in *_columns) but... It needs more eyes.